src/main/java/com/aws/greengrass/shadowmanager/sync/strategy/BaseSyncStrategy.java

```diff
@@ -41,6 +39,7 @@
 public abstract class BaseSyncStrategy implements SyncStrategy {
     private static final Logger logger = LogManager.getLogger(BaseSyncStrategy.class);
     static final String SYNC_EVENT_TYPE = SYNC.code();
+    static final int THREAD_END_WAIT_TIME_SECONDS = 10;
```

Comment:
It looks like this changed from 30 seconds to a 10 second timeout, is there data supporting the quicker timeout?

Reply:
there's also no data supporting the longer one - 30 seconds was also an arbitrary number.

In the logs on one of the failed acceptance tests, syncing was stopped, and the IPC publisher thread was blocked for 30 seconds (due to it waiting for a latch that would never countdown)

I'd rather not block for 30 seconds - a single sync request shouldn't take 10s - except for an incredibly slow network